### DISCUSSION TITLE
How to get the path for shared articafacts of a package?

### DISCUSSION THREAD

**Discussion by s1n7ax** [mason-org/mason.nvim#1553] [Q&A] [answered]
There are API like  `registry.get_pkg_path` and `package:get_install_path` however I couldn't find an API to get the path to shared artifacts.

My requirement is to get  `.jar`s  from `java-test` package instead of looking for `.jar`s inside installed path.

**Comment by williamboman** [comment, marked as answer]
Reaching into the internal package structure is not recommended and not supported as this may change in-between versions. Instead, these artifacts are also available at a stable location inside the `bin/` (executables), `share/` (architecture-independent files), and `opt/` (rarely used for architecture-dependent files) directories. `.jar` files will be found inside the `share/` directory. The recommended way to access these are by expanding the `$MASON` environment variable which mason.nvim sets up during setup.

In the case of java-test, this could for example look like:

```lua
local java_test_jar = vim.fn.glob "$MASON/share/java-test/com.microsoft.java.test.plugin-*.jar"
-- /Users/williamboman/.local/share/nvim/mason/share/java-test/com.microsoft.java.test.plugin-0.39.0.jar

-- do something with the file path to the jar…
```

Note that the above example uses `:h glob()` which will return the empty string `""` if the file doesn't exist (i.e. `java-test` is not yet installed).
